routers/routers.go

```diff
@@ -10,9 +10,20 @@ import (
 	"github.com/atmiguel/cerealnotes/paths"
 )
 
+type routeHandler struct {
```

Comment:
This "embeds" a type which is super cool. It allows callers to access the methods of the embedded type without need endless wrapper methods.

Reply:
I consider it to be more like classical inheritance than prototype inheritance. Type embedding _is_ like prototype inheritance, but only because prototype inheritance is like classical inheritance. Off the top of my head I can't think of anything similar between type embedding and prototype inheritance that isn't also similar between type embedding and classical inheritance, so comparing it to prototype inheritance is a bit misleading.

For example, with prototype inheritance you can dynamically add/remove functions to the prototype at runtime and all objects will be affected, even objects that were instantiated prior to messing with the prototype. Type embedding doesn't have anything like that